test/versioning/semver-ruby.spec.js

```diff
@@ -0,0 +1,405 @@
+      expect(semverRuby.isValid('! 1.0.0-beta')).toBeFalsy();
+      expect(semverRuby.isValid('& 1.0.0.beta')).toBeFalsy();
+    });
+  });
+
+  describe('.isSingleVersion', () => {
+    it('returns true when version is single', () => {
+      expect(semverRuby.isSingleVersion('1')).toBeTruthy();
+      expect(semverRuby.isSingleVersion('1.2')).toBeTruthy();
+      expect(semverRuby.isSingleVersion('1.2.1')).toBeTruthy();
+
+      expect(semverRuby.isSingleVersion('=1')).toBeTruthy();
+      expect(semverRuby.isSingleVersion('=1.2')).toBeTruthy();
+      expect(semverRuby.isSingleVersion('=1.2.1')).toBeTruthy();
+
+      expect(semverRuby.isSingleVersion('= 1')).toBeTruthy();
+      expect(semverRuby.isSingleVersion('= 1.2')).toBeTruthy();
+      expect(semverRuby.isSingleVersion('= 1.2.1')).toBeTruthy();
+
+      expect(semverRuby.isSingleVersion('1.2.1.rc1')).toBeTruthy();
+      expect(semverRuby.isSingleVersion('1.2.1-rc.1')).toBeTruthy();
+
+      expect(semverRuby.isSingleVersion('= 1.2.0.alpha')).toBeTruthy();
+      expect(semverRuby.isSingleVersion('= 1.2.0-alpha')).toBeTruthy();
+    });
+
+    it('returns false when version is multiple', () => {
+      expect(semverRuby.isSingleVersion('!= 1')).toBeFalsy();
+      expect(semverRuby.isSingleVersion('> 1.2')).toBeFalsy();
+      expect(semverRuby.isSingleVersion('< 1.2.1')).toBeFalsy();
+      expect(semverRuby.isSingleVersion('>= 1')).toBeFalsy();
+      expect(semverRuby.isSingleVersion('<= 1.2')).toBeFalsy();
+      expect(semverRuby.isSingleVersion('~> 1.2.1')).toBeFalsy();
+    });
+
+    it('returns false when version is invalid', () => {
+      expect(semverRuby.isSingleVersion()).toBeFalsy();
+      expect(semverRuby.isSingleVersion('')).toBeFalsy();
+      expect(semverRuby.isSingleVersion(null)).toBeFalsy();
+      expect(semverRuby.isSingleVersion('tottally-not-a-version')).toBeFalsy();
+    });
+  });
+
+  describe('.getNewValue', () => {
+    it('returns pinned version for pin strategy', () => {
+      [
+        ['1.2.3', '1.0.3', 'pin', '1.0.0', '1.2.3'],
+        ['1.2.3', '= 1.0.3', 'pin', '1.0.0', '1.2.3'],
+        ['1.2.3', '!= 1.0.3', 'pin', '1.0.0', '1.2.3'],
+        ['1.2.3', '> 1.0.3', 'pin', '1.0.0', '1.2.3'],
+        ['1.2.3', '< 1.0.3', 'pin', '1.0.0', '1.2.3'],
+        ['1.2.3', '>= 1.0.3', 'pin', '1.0.0', '1.2.3'],
+        ['1.2.3', '<= 1.0.3', 'pin', '1.0.0', '1.2.3'],
+        ['1.2.3', '~> 1.0.3', 'pin', '1.0.0', '1.2.3'],
+        ['1.2.3', '~> 4.7, >= 4.7.4', 'pin', '1.0.0', '1.2.3'],
+      ].forEach(([expected, ...params]) => {
+        expect(semverRuby.getNewValue(...params)).toEqual(expected);
+      });
+    });
+
+    it('returns bumped version for bump strategy', () => {
+      [
+        ['1.2.3', '1.0.3', 'bump', '1.0.0', '1.2.3'],
+        ['= 1.2.3', '= 1.0.3', 'bump', '1.0.0', '1.2.3'],
+        ['!= 1.0.3', '!= 1.0.3', 'bump', '1.0.0', '1.2.3'],
+        ['> 1.0.3', '> 1.0.3', 'bump', '1.0.0', '1.2.3'],
```

Comment:
I'm not sure if people would expect this to be "bumped" to `> 1.2.2` ? It would be a pretty weird config though.

Reply:
Thanks, fixed